### PR TITLE
Fix bugs: changing time step after restart and tracer tanh model

### PR DESCRIPTION
**Pull request by oguevremont**
### Description

There were two issues that appeared when using the tracer physics with an immersed solid:
- Evaluation of the diffusivity model `immersed solid tanh` was too costly
- Changing the `time step` after restarting was impossible

### Solution

- The diffusivity model now uses the optimizations from the Shape class, that allow to evaluate the distance using a cell guess; this makes RBF and composite distance evaluations magnitudes faster.
- Time step change after a restart is now allowed when `adapt==false`. This behavior is not allowed when `adapt==true`, as then we desire a continuity in time step evolution. When using fixed time step though, users who change the value after a restart apparently aim for this behavior.

### Testing

All tests still pass.

### Documentation

No change needed.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Copyright headers are present and up to date
- [x] Lethe documentation is up to date
- [x] Fix has unit test(s) (preferred) or application test(s), and restart files are in the generator folder
- [x] The branch is rebased onto master
- [x] Changelog (CHANGELOG.md) is up to date
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent

Pull request related list:
- [x] Labels are applied
- [ ] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [ ] If the fix is temporary, an issue is opened
- [ ] The PR description is cleaned and ready for merge

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,13 @@
 All notable changes to the Lethe project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## [Master] - 2024-11-07
+
+### Fix
+
+- MINOR Time step can now be changed after a restart if adaptive time stepping is disabled. [#1343](https://github.com/chaos-polymtl/lethe/pull/1343)
+- MINOR Application of the immersed solid tanh diffusivity model is now faster thanks to using optimizations from the Shape class. [#1343](https://github.com/chaos-polymtl/lethe/pull/1343)
```
Comment (lpsaavedra) [inline CHANGELOG.md:11]:
```suggestion
- MINOR Time step can now be changed after a restart if adaptive time stepping is disabled. [#1343](https://github.com/chaos-polymtl/lethe/pull/1343)

- MINOR Application of the immersed solid tanh diffusivity model is now faster thanks to using optimizations from the Shape class. [#1343](https://github.com/chaos-polymtl/lethe/pull/1343)
```

**Review by lpsaavedra — CHANGES_REQUESTED** [review]
I have a few comments and doubts regarding these new features

File: source/core/simulation_control.cc [hunk 2]
```diff
@@ -263,7 +263,17 @@ SimulationControlTransient::integrate()
...
+void
+SimulationControlTransient::read(const std::string &prefix)
+{
+  SimulationControl::read(prefix);
 
+  // Fix the time step to the new provided value.
+  // We understand that users providing this value when adaptive time stepping
+  // is not enabled means a time step change is the desired effect.
+  if (!adapt)
+    set_current_time_step(initial_time_step);
```
[the remark this answers is not in the document]
Comment (oguevremont) [reply source/core/simulation_control.cc:275]: Yes it is risky business, but the way I use it is that I solve NS with small time steps, until the solution is steady, then I solve only tracer with large time steps after a restart (before and after the restart the time step is fixed). I should write this somewhere so it is clearer, thanks for pointing it out

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,12 +3,20 @@
 All notable changes to the Lethe project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## [Master] - 2024-11-18
+
+### Fix
```
Comment (lpsaavedra) [inline CHANGELOG.md:8]:
```suggestion
### Fixed
```

**Review by lpsaavedra — APPROVED** [review]
I just have one additional minor comment...

**Review by blaisb — APPROVED** [review]
Much better this way I like it

**Pull request merged**